components/python/isort/Makefile

```diff
@@ -0,0 +1,85 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2021 Gary Mills
+#
+
+BUILD_STYLE=	setup.py
+BUILD_BITS=	NO_ARCH
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		isort
+COMPONENT_VERSION=	5.9.3
+# COMPONENT_REVISION=	0
+COMPONENT_PROJECT_URL=	https://github.com/PyCQA/isort
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	\
+	sha256:0ad5e0ec066302c7aece422b2675b81abe7e417dbf64b18438ca8b79219540dd
+COMPONENT_ARCHIVE_URL=	https://github.com/PyCQA/$(COMPONENT_NAME)/archive/refs/tags/$(COMPONENT_VERSION).tar.gz
+COMPONENT_BUGDB=	python-mod/isort
+
+# Set python version used by this product
+PYTHON_VERSION=		3.9
+PYTHON_VERSIONS=	3.9
+
+include $(WS_MAKE_RULES)/common.mk
+
+# Copy in a minimal setup.py
+COMPONENT_PRE_BUILD_ACTION = $(CP) OI/setup.py $(COMPONENT_SRC)
+
+# The tests are run using python 3.9 only and require that the
+# python-39 package is installed (does not have to be the default python).
+# The pytest-39, hypothesis-39, and sortedcontainers-39 packages must
+# also be installed
+
```

Comment:
We usually add REQUIRED_PACKAGES += statements for test dependencies with an appropriate comment (eg. # Manually added test dependencies) but I think such a comment is also ok.
What do you think?

Reply:
The problem is that pytest and a few other packages are not dependancies of the packages described by the P5M files;  Instead, they are dependancies of the test modules.  Adding them to the REQUIRED_PACKAGES list will have no effect, although it might provide a clue for the developer.  Comments are the best we can do just now.  The automatic package install always fails for me because I do the builds as an unprivileged user.

I suppose a different variable, such as TEST_REQUIRED_PACKAGES or REQUIRED_TEST_PACKAGES might be an improvement.  Then, the test procedure could at least ensure that packages from that list are installed.

Reply:
Yes, that sounds reasonable. We only need to find someone who will provide such an enhancement.